src/ripple/app/misc/AmendmentTable.h

```diff
@@ -154,10 +211,26 @@ class AmendmentTable
     doVoting (Ledger::ref lastClosedLedger, SHAMap::ref initialPosition) = 0;
 };
 
-std::unique_ptr<AmendmentTable>
-make_AmendmentTable (std::chrono::seconds majorityTime, int majorityFraction,
-    beast::Journal journal);
+namespace AmendmentTableDetail
+{
+/**
+   AppApiFacade is used to insert mock objects into the amendment
+   table class while unit testing.
+ */
+enum class AppApiFacade
+{
+    useApp,
+    useMock
```

Comment:
There's no need to proliferate identifiers used in such a narrow scope, when a bool suffices. Although if its in a detail namespace I am less concerned. Its still overly verbose. But looking at the bigger picture, a bool is inferior to an abstract interface. The implementation should not be aware of whether or not it is being passed a mock interface.